lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -187,10 +220,35 @@ def _on_legacy_relation_joined(self, event):
             "replset": self.charm.app.name,
         }
 
-        # reactive charms set relation data on "the current unit"
+        # reactive charms set relation data on "the current unit" the reactive mongodb charm sets
+        # the relation data for all units, hence all units setting the relation data and not just
+        # the leader
         relation = self.model.get_relation(REL_NAME, event.relation.id)
         relation.data[self.charm.unit].update(updates)
 
+    def get_users_from_legacy_relations(self):
+        """Return usernames for legacy relation."""
+        relations = self.model.relations[LEGACY_REL_NAME]
+        logger.debug(relations)
+        return set([self._get_username_from_relation_id(relation.id) for relation in relations])
```

Comment:
Just a fun tip, you can use set-comprehension here:

`{self._get_username_from_relation_id(relation.id) for relation in relations}`